src/core/algorithms/gfd/gfd_miner.cpp

```diff
@@ -0,0 +1,654 @@
+    bool result = false;
+    std::map<vertex_t, std::set<vertex_t>> maps = {};
+    for (auto& embedding : embeddings) {
+        if (!Satisfied(gfd.GetPremises(), graph, gfd.GetPattern(), embedding)) {
+            continue;
+        }
+        result = true;
+
+        if (!Satisfied(gfd.GetConclusion(), graph, gfd.GetPattern(), embedding)) {
+            return false;
+        }
+
+        BGL_FORALL_VERTICES_T(v, gfd.GetPattern(), graph_t) {
+            if (maps.find(v) != maps.end()) {
+                maps.at(v).insert(embedding.at(v));
+            } else {
+                std::set<vertex_t> vertices = {embedding.at(v)};
+                maps.emplace(v, vertices);
+            }
+        }
+    }
+
+    std::size_t min_size = boost::num_vertices(graph);
+    for (auto& [v, image] : maps) {
+        min_size = std::min(image.size(), min_size);
+    }
+    return result && (min_size >= sigma);
+}
+
+std::vector<Literal> GenerateLiterals(graph_t const& pattern,
+                                      std::map<std::string, Info> const& attrs_info) {
+    std::vector<size_t> indices(boost::num_vertices(pattern));
+    std::iota(std::begin(indices), std::end(indices), 0);
+    std::vector<std::vector<size_t>> pairs = GetSubsets<size_t>(indices, 2);
+
+    std::vector<Literal> result = {};
+    for (auto const& index_pair : pairs) {
+        size_t fst = index_pair.at(0);
+        size_t snd = index_pair.at(1);
+        std::string fst_label = pattern[boost::vertex(fst, pattern)].attributes.at("label");
+        std::string snd_label = pattern[boost::vertex(snd, pattern)].attributes.at("label");
+
+        for (auto const& fst_name : attrs_info.at(fst_label)) {
+            Token fst_token = {fst, fst_name.first};
+            for (auto const& snd_name : attrs_info.at(snd_label)) {
+                Token snd_token = {snd, snd_name.first};
+                result.push_back({fst_token, std::move(snd_token)});
+            }
+        }
+    }
+    for (size_t i = 0; i < boost::num_vertices(pattern); ++i) {
+        std::string label = pattern[boost::vertex(i, pattern)].attributes.at("label");
+        for (auto& name : attrs_info.at(label)) {
+            Token name_token = {i, name.first};
+            for (auto& value : name.second) {
+                Token value_token = {-1, value};
+                Literal l = {name_token, std::move(value_token)};
+                result.push_back(std::move(l));
+            }
+        }
+    }
+    return result;
+}
+
+std::size_t Support(graph_t const& graph, Embeddings const& embeddings, graph_t const& pattern) {
+    std::map<vertex_t, std::set<vertex_t>> result = {};
+    BGL_FORALL_VERTICES_T(v, pattern, graph_t) {
+        for (auto& emb : embeddings) {
+            if (result.find(v) != result.end()) {
+                result.at(v).insert(emb.at(v));
+            } else {
+                std::set<vertex_t> vertices = {emb.at(v)};
+                result.emplace(v, vertices);
```

Comment:
Please, use move semantics everywhere else
```suggestion
                result.emplace(v, std::move(vertices));
```